src/rpc_server.rs

```diff
@@ -2013,6 +2011,11 @@ impl NeptuneRPCServer {
                 return Err(e.into());
             }
         };
+        let mut transaction = transaction_creation_artifacts.transaction;
+        let transaction_details = transaction_creation_artifacts
+            .details
+            .expect("details should be some when configured to track details");
+        let maybe_change_output = transaction_creation_artifacts.change_output;
         drop(state);
```

Comment:
Can you drop the state earlier than this? It seems like it's not needed in the added lines.